Jellyfin.Networking/Manager/NetworkManager.cs

```diff
@@ -0,0 +1,1283 @@
+            }
+        }
+
+        /// <inheritdoc/>
+        public bool IsExcludedInterface(IPAddress address)
+        {
+            lock (_intLock)
+            {
+                if (_bindExclusions.Count > 0)
+                {
+                    return _bindExclusions.Contains(address);
+                }
+
+                return false;
+            }
+        }
+
+        /// <inheritdoc/>
+        public NetCollection GetFilteredLANSubnets(NetCollection? filter = null)
+        {
+            lock (_intLock)
+            {
+                if (filter == null)
+                {
+                    return NetCollection.AsNetworks(_lanSubnets.Exclude(_excludedSubnets));
+                }
+
+                return _lanSubnets.Exclude(filter);
+            }
+        }
+
+        /// <inheritdoc/>
+        public bool IsValidInterfaceAddress(IPAddress address)
+        {
+            lock (_intLock)
+            {
+                return _interfaceAddresses.Contains(address);
+            }
+        }
+
+        /// <inheritdoc/>
+        public bool TryParseInterface(string token, out NetCollection? result)
+        {
+            result = null;
+            if (string.IsNullOrEmpty(token))
+            {
+                return false;
+            }
+
+            if (_interfaceNames != null && _interfaceNames.TryGetValue(token.ToLower(CultureInfo.InvariantCulture), out int index))
+            {
+                result = new NetCollection();
+
+                _logger.LogInformation("Interface {0} used in settings. Using its interface addresses.", token);
+
+                // Replace interface tags with the interface IP's.
+                foreach (IPNetAddress iface in _interfaceAddresses)
+                {
+                    if (Math.Abs(iface.Tag) == index &&
+                        ((IsIP4Enabled && iface.Address.AddressFamily == AddressFamily.InterNetwork) ||
+                         (IsIP6Enabled && iface.Address.AddressFamily == AddressFamily.InterNetworkV6)))
+                    {
+                        result.Add(iface);
+                    }
+                }
+
+                return true;
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Reloads all settings and re-initialises the instance.
+        /// </summary>
+        /// <param name="config"><seealso cref="ServerConfiguration"/> to use.</param>
+        public void UpdateSettings(ServerConfiguration config)
+        {
+            if (config == null)
+            {
+                throw new ArgumentNullException(nameof(config));
+            }
+
+            IsIP4Enabled = Socket.OSSupportsIPv6 && config.EnableIPV4;
+            IsIP6Enabled = Socket.OSSupportsIPv6 && config.EnableIPV6;
+            TrustAllIP6Interfaces = config.TrustAllIP6Interfaces;
+            UdpHelper.EnableMultiSocketBinding = config.EnableMultiSocketBinding;
+
+            if (string.IsNullOrEmpty(MockNetworkSettings))
```

Comment:
I don't like having test/debug specific stuff in the main classes. Can this be placed somewhere else somehow?

Reply:
I know, the only other solution is to create a entire mock of the interface structure - which is a massive task.  
Implemented this as a way of being able to replicate different network environments. It's used in the testing units.

Did look to see if a mock existed online - but couldn't find anything.